src/coreclr/jit/llvm_types.h

```diff
@@ -12,8 +12,30 @@
 
 struct TypeDescriptor
 {
-    unsigned              fieldCount;
-    CORINFO_FIELD_HANDLE* fields;
+private:
+    unsigned              m_fieldCount;
+    CORINFO_FIELD_HANDLE* m_fields;
+    unsigned              m_isExplicitLayout;
+
+    TypeDescriptor()
+    {
+    }
+
+public:
+    unsigned getFieldCount()
+    {
+        return m_fieldCount;
+    }
+
+    CORINFO_FIELD_HANDLE getField(unsigned index)
+    {
+        return m_fields[index];
+    }
+
+    unsigned getIsExplicitLayout()
+    {
+        return m_isExplicitLayout;
+    }
```

Comment:
The condition here is: `LayoutKind.Explicit || HasSizeMetadata`, i. e. the following struct should also be treated as one with significant padding:
```cs
[StructLayout(LayoutKind.Sequential, Size = 10)]
struct S
{
    public byte Field;
}
```
This would also imply renaming this (and related interface code) to something like `hasSignificantPadding`.

Reply:
Thanks